src/prefect/__init__.py

```diff
@@ -23,8 +23,8 @@
 if sys.version_info < (3, 8):
     warnings.warn(
         (
-            "Prefect will drop support for Python 3.7 when it reaches end-of-life on 27"
-            " Jun 2023. To use new versions of Prefect after that date, you will need"
+            "Prefect dropped support for Python 3.7 when it reached end-of-life"
```

Comment:
I'm not sure if this warning is ever reachable, because `setup.py` now requires 3.8? No harm to leave this as-is, of course :)